### PR TITLE
Dockerize application

### PR DESCRIPTION
**Pull request by mccodav1**
This change Dockerizes the application. The README has been updated with details on how to run the application via Docker. The application is intended to be served via docker-compose (or eventually, Docker Swarm) in a production environment.

When running via Docker Compose, the project utilizes a Postgres database and is served via Gunicorn. However, one can also still run the application via Django's development server and an SQLite database: see the README for details.

Essentially, the docker-compose file handles selecting the database automatically; if you choose to run the application the way we have in the past via `python manage.py`, the SQLite database will be used automatically. This means that no changes should be apparent if not using Docker, but Docker will be the preferred modality moving forward.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,56 @@ This repository contains the Poste backend server application / API as a Python
...
+
+>https://www.docker.com/get-started/
+
+4. Run the application via Docker using Docker Compose. Optional Make target provided for convenience.
+
+`docker compose up --build -d` or `make up` 
+
+5. The application will be running locally:
+ 
+>http://localhost
```
Comment (kmgehrma) [inline README.md:30]:
```suggestion
>http://localhost:8000
```

**Review by techbykim — CHANGES_REQUESTED** [review]
When I ran via `make up` I received the following error:
image

When I ran via `docker compose up --build -d` I also ran into the following error:
image

It look like the Postgres container is being created properly but there is an issue with the Django backend:
image

**Comment by mccodav1** [comment]
> When I ran via `make up` I received the following error: image When I ran via `docker compose up --build -d` I also ran into the following error:

This is likely an issue with line endings, which differs by system and git likes to change up on its own. I imagine you're on Mac? If so, try either changing the line endings in a text editor or via git, or copy the entire entry point into a new file (which will create a new file with line endings that work in your system)

Not a simple fix to commit because Windows and Mac/Linux want different line endings, so may take some tweaking locally.



> It look like the Postgres container is being created properly but there is an issue with the Django backend:

The other services depend on Postgres operating in a healthy state. It appears Postgres won't stand up correctly for you due to the line endings discussed above; if you fix this locally it should be able to execute it's entry point and load to a healthy state, allowing the other containers to load

File: README.md (hunk 1 above)
Comment (mccodav1) [reply README.md:30]: http://localhost/ is correct. It should be accessed by port 80 (and eventually 443 via https) so specifying the port is not appropriate

**Comment by mccodav1** [comment]
@kmgehrma Requesting re-review or approval. I'm able to run this on multiple computers; for those using Mac / Linux, local modifications may be necessary (but committing those changes would be moot, as it then wouldn't work for those using Windows). Seems like something that will just take some local tweaking.

image

**Review by techbykim — APPROVED** [review]

**Comment by kmgehrma** [comment]
conditionally approving because I have Mac not Windows

**Comment by fmehdad97** [comment]
@kmgehrma I have it running on MacOS currently. I did not face the Makefile issue you mentioned, but running the Make command did lead me to the entrypoint issue. 
Adding the following line of code fixed it for me though:
`RUN chmod +x /usr/src/app/deploy/entrypoint.sh`
You want to add in the file named Dockerfile, in the line before `ENTRYPOINT ["/usr/src/app/deploy/entrypoint.sh"]`


@mccodav1 Can you try adding this line on Windows too and ensure things dont break? (Let me know if you are unable to and I'll fire up and run it on my Windows machine). Based on my understanding, I don't think it should break things. If it doesn't, could you please merge this small change to dev branch on the backend?

**Pull request merged**